### PR TITLE
docs: Add Docker Compose Setup for ServerlessLLM

### PR DESCRIPTION
**Pull request by future-xy**
## Description
This PR introduces a Docker Compose setup to simplify the startup process:
- Added a `docker-compose.yml` file.
- Updated the Docker quick start guide with new instructions for using Docker Compose.

## Motivation
This update makes it easier to launch a ServerlessLLM cluster on a local GPU machine for development and testing. It also completes a roadmap item outlined in #113.

## Type of Change
- [x] Documentation update

## Checklist
- [x] I have read the [CONTRIBUTION](https://github.com/ServerlessLLM/ServerlessLLM/blob/main/CONTRIBUTING.md) guide.
- [x] I have updated the documentation (if applicable).

### CODE REVIEW

File: docs/stable/getting_started/docker_quickstart.md [hunk 1]
```diff
@@ -4,92 +4,52 @@ sidebar_position: 2
...
+1. **Docker**: Installed on your system. You can download it from [here](https://docs.docker.com/get-docker/).
+2. **ServerlessLLM CLI**: Installed on your system. You can install it using `pip install serverless-llm`.
+1. **GPUs**: At least 2 NVIDIA GPUs are necessary. If you have more GPUs, you can adjust the `docker-compose.yml` file accordingly.
+2. **NVIDIA Docker Toolkit**: This allows Docker to use NVIDIA GPUs. Follow the installation guide [here](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html).
 
-## Run ServerlessLLM using Docker
+## Run ServerlessLLM using Docker Compose
 
-First, let's start a local Docker-based ray cluster to run ServerlessLLM.
+We will use docker-compose to simplify the setup of ServerlessLLM. The `docker-compose.yml` file is located in the `examples/docker_compose/` directory of the ServerlessLLM repository.
```
Comment (SiyangShao) [inline docs/stable/getting_started/docker_quickstart.md:20]: Currently it's under `examples/docker_cluster/` directory.

File: docs/stable/getting_started/docker_quickstart.md [hunk 2]
```diff
@@ -4,92 +4,52 @@ sidebar_position: 2
...
+### Step 1: Clone the ServerlessLLM Repository
 
-Run the following commands to build the Docker images:
+If you haven't already, clone the ServerlessLLM repository:
 
 ```bash
-docker build . -t serverlessllm/sllm-serve
-docker build -f Dockerfile.worker . -t serverlessllm/sllm-serve-worker
+git clone https://github.com/serverless-llm/serverlessllm.git
+cd serverlessllm/examples/docker_compose/
```
Comment (SiyangShao) [inline docs/stable/getting_started/docker_quickstart.md:28]: Same, now it's under `examples/docker_cluster`

File: docs/stable/getting_started/docker_quickstart.md [hunk 3]
```diff
@@ -4,92 +4,52 @@ sidebar_position: 2
...
-  serverlessllm/sllm-serve-worker
-
-docker run -d --name ray_worker_1 \
-  --runtime nvidia \
-  --network sllm \
-  --gpus '"device=1"' \
-  --env WORKER_ID=1 \
-  --mount type=bind,source=$MODEL_FOLDER,target=/models \
-  serverlessllm/sllm-serve-worker
+docker-compose up -d --build
```
Comment (SiyangShao) [inline docs/stable/getting_started/docker_quickstart.md:47]: Seems that in current version, it should be `docker compose` compared with `docker-compose`? ref: https://docs.docker.com/compose/releases/migrate/

File: docs/stable/getting_started/docker_quickstart.md [hunk 4]
```diff
@@ -143,22 +103,7 @@ sllm-cli delete facebook/opt-1.3b
...
-
-```bash
-docker exec ray_head sh -c "ray stop"
-docker exec ray_worker_0 sh -c "ray stop"
-docker exec ray_worker_1 sh -c "ray stop"
-
-docker stop ray_head ray_worker_0 ray_worker_1
-docker rm ray_head ray_worker_0 ray_worker_1
-docker network rm sllm
+docker-compose down
```
Comment (SiyangShao) [inline docs/stable/getting_started/docker_quickstart.md:108]: Same. `docker compose down` may be better.

**Review by SiyangShao — CHANGES_REQUESTED** [review]

**Comment by JustinTong0323** [comment]
As the default mem_pool_size for sllm-store-server is 4, which is relative small and difficult to handle large model, I think there maybe need a option for adjusting it when using docker to deploy the sllm.

**Comment by future-xy** [comment]
> As the default mem_pool_size for sllm-store-server is 4, which is relative small and difficult to handle large model, I think there maybe need a option for adjusting it when using docker to deploy the sllm.

sounds good, will fix later

**Review by SiyangShao — APPROVED** [review]
LGTM. BTW, will the configuration for mem_pool_size be in another PR or in this PR?

**Comment by future-xy** [comment]
> LGTM. BTW, will the configuration for mem_pool_size be in another PR or in this PR?

It will be in a separate PR, as it involves modifying the entrypoint and requires additional testing.

**Pull request merged**